fineract-core/src/main/java/org/apache/fineract/portfolio/savings/data/SavingsAccountTransactionData.java

```diff
@@ -144,9 +147,12 @@ public boolean isOverdraftInterestAndNotReversed() {
         return this.transactionType.isIncomeFromInterest() && isNotReversed();
     }
 
+    public boolean isCredit() {
+        return transactionType.getTransactionTypeEnum().isCredit() && isNotReversed() && !isReversalTransaction();
```

Comment:
I was wondering if there is a reversal transaction which is DEBIT, neither the isCredit, neither the isDebit function returns true. Is this valid?

Reply:
Again I would love to clean this up, but wanted to change as less as I could. (isDebit should return true or false independently from the reversal/reversed status and the place of usage should decide what to do with these transactions)
But if the reversed transactions are not considered to be neither credit nor debit, then the reversal transaction (which reversed it) should behave the same way, otherwise the balance would not be correct.